BudgetSquirrel.Api/Services/Implementations/AsyncQueryService.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using BudgetSquirrel.Business;
+using BudgetSquirrel.Data.EntityFramework;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Query;
+
+namespace BudgetSquirrel.Api.Services.Implementations
+{
+  public class AsyncQueryService : IAsyncQueryService
+  {
+    private readonly BudgetSquirrelContext context;
+
+    public AsyncQueryService(BudgetSquirrelContext context)
+    {
+      this.context = context;
+    }
+
+    public IQueryable<T> Include<T, TProperty>(IQueryable<T> source, Expression<Func<T, TProperty>> include) where T : class
+    {
+      if (source is DbSet<T>)
+      {
+        DbSet<T> includable = ((DbSet<T>) source);
+        return includable.Include(include);
+      }
+      else
+      {
+        throw new InvalidOperationException("source must be a DbSet, not " + source.GetType().Name);
+      }
+    }
+
+    public Task<T> SingleOrDefaultAsync<T>(IQueryable<T> source, Expression<Func<T, bool>> predicate)
+    {
+      return source.SingleOrDefaultAsync(predicate);
+    }
+
+    public Task<bool> AnyAsync<T>(IQueryable<T> source, Expression<Func<T, bool>> predicate)
+    {
+      return source.AnyAsync(predicate);
+    }
+
+    public Task SaveChangesAsync()
+    {
+      return this.SaveChangesAsync();
+    }
```

Comment:
Why don't we await any of the methods with the name async in them and why don't we just call the method that these methods are calling instead. 

Am I making any sense lol, don't think I am.

Reply:
See my response above to singleOrDefaultAsync